cli/cmd/resources/odiglet.go

```diff
@@ -588,7 +671,7 @@ func (a *odigletResourceManager) InstallFromScratch(ctx context.Context) error {
 
 	// before creating the daemonset, we need to create the service account, cluster role and cluster role binding
 	resources = append(resources,
-		NewOdigletDaemonSet(a.ns, a.odigosVersion, a.config.ImagePrefix, a.managerOpts.ImageReferences.OdigletImage, a.odigosTier, a.config.OpenshiftEnabled,
+		NewOdigletDaemonSet(ctx, a.client, a.ns, a.odigosVersion, a.config.ImagePrefix, a.managerOpts.ImageReferences.OdigletImage, a.odigosTier, a.config.OpenshiftEnabled,
```

Comment:
the `client` here is actually meant to be used for `ApplyResources`.
The `NewOdigletDaemonSet` function only cares if the file is present or not, so you can consider hoisting this query to a prior stage and just pass a bool to `odigletResourceManager` and indicate if offset file should be injected or not.

This will help keep the resource manager `New...` functions as pure functions, that cannot have errors or side effect, and will not `os.Exit` the process.

Reply:
I created a new resource function `NewOdigletGoOffsetsConfigMap`, which does still check for an existing map (to not overwrite the data), and handles errors better through the current InstallFromScratch flow. Let me know what you think